Yatzy.Tests/GameTests.cs

```diff
@@ -0,0 +1,149 @@
+using Moq;
+using Yatzy_Kata;
+using Yatzy_Kata.Data;
+using Yatzy_Kata.Factories;
+using Yatzy_Kata.Interfaces;
+using Yatzy_Kata.Outcomes;
+using Yatzy_Kata.Strategies;
+
+namespace Yatzy_Test;
+
+
+public class GameTests
+{
+    private readonly List<Player> _players;
+    private readonly Game _game;
+    private readonly Mock<IRandom> _mockPlayer1RandomDieRoll;
+    private readonly Mock<IRandom> _mockPlayer2RandomDieRoll;
+    private readonly Mock<IReader> _mockReader;
+
+    public GameTests()
+    {
+        _mockReader = new Mock<IReader>();
+        _mockPlayer1RandomDieRoll = new Mock<IRandom>();
+        _mockPlayer2RandomDieRoll = new Mock<IRandom>();
+        _mockPlayer1RandomDieRoll.Setup(diceHand => diceHand.GetDiceHand()).Returns(new List<int> { 5, 5, 5, 5, 5 });
+        _mockPlayer2RandomDieRoll.Setup(diceHand => diceHand.GetDiceHand()).Returns(new List<int> { 4, 4, 2, 3, 3 });
+        _players = new List<Player> { new Player("bundy", new ComputerStrategy(_mockPlayer1RandomDieRoll.Object)), new Player("john", new ComputerStrategy(_mockPlayer2RandomDieRoll.Object)) };
+        _game = new Game(_players, new RoundFactory(new TurnFactory()));
+        
+    }
+    
+    [Fact]
+    public void GivenPlayGameIsCalled_WhileEveryPlayerPLaysTillNoCategoriesLeft_ThenAMaximumOfThirteenRoundsArePlayed()
+    {
+        //Arrange
+        var stringWriter = new StringWriter();
+        Console.SetOut(stringWriter);
+
+        //Act
+        _game.PlayGame();
+        
+        //Assert
+        Assert.Contains(ConsoleMessages.RoundCount(13), stringWriter.ToString());
+        Assert.DoesNotContain(ConsoleMessages.RoundCount(14), stringWriter.ToString());
+    }
+    
+    [Fact]
+    public void GivenPlayGameIsCalled_ThenGamePromptsWelcomeMessage()
+    {
+        //Arrange
+        var stringWriter = new StringWriter();
+        Console.SetOut(stringWriter);
+
+        //Act
+        _game.PlayGame();
+        
+        //Assert
+        Assert.Contains(ConsoleMessages.Welcome, stringWriter.ToString());
+    }
+
+    [Fact]
+    public void GivenPlayGameIsCalled_WhileEveryPlayerPLaysTillNoCategoriesLeft_ThenThePlayerWithTheMostPointsWillBePromptedAtTheEndOfTheGame()
+    {
+        //Arrange
+        var stringWriter = new StringWriter();
+        Console.SetOut(stringWriter);
+
+        //Act
+        _game.PlayGame();
+        
+        //Assert
+        Assert.Contains(ConsoleMessages.Winner(_players[0]), stringWriter.ToString());
+    }
+    
+    [Fact]
+    public void GivenPlayGameIsCalled_WhileEveryPlayerPLaysTillNoCategoriesLeft_ThenPLayersWillHaveNoCategoriesLeft()
```

Comment:
Great to see some descriptive names here, but since the Given statement is the same for each method  maybe we can rename the file to that given statement and remove the duplication?